### ISSUE TITLE
Add ability to specify a key prefix for ext_pillar S3

### ISSUE THREAD

**Issue by EliRibble** [saltstack/salt#22472]
It would be useful to avoid the workaround in https://github.com/saltstack/salt/issues/22471 if we could specify a key prefix to use when downloading pillar data from S3. This would effectively allow the pillar data to be in a 'subdirectory' of the S3 bucket.

Essentially you could do this:

```
ext_pillar:
  - s3:
    bucket: my-bucket
    keyid: *redacted*
    key: *redacted*
    multiple_env: False
    verify_ssl: True
    environment: prod
    prefix: pillars/prod
```

And salt would pull the pillar data from `s3://my-bucket/pillars/prod/top.sls`. This could also work in multiple_env mode as

```
ext_pillar:
  - s3:
    bucket: my-bucket
    keyid: *redacted*
    key: *redacted*
    multiple_env: True
    verify_ssl: True
    environment: prod
    prefix: deployment/pillars
```

and would pull from 

```
s3://my-bucket/deployment/pillars/base/*
s3://my-bucket/deployment/pillars/prod/*
```

**Comment by rallytime** [comment]
Thanks for the report and for the pull request. Once that gets merged, is this issue ok to close?

**Comment by EliRibble** [comment]
I just added a new pull request, I was missing an essential line to make it work.

**Comment by rallytime** [comment]
Awesome! Thanks @EliRibble! Is this good to go once the new PR is merged?

**Comment by EliRibble** [comment]
Okay, had to add in some changes to the caching, but it's working now in my environment.

**Comment by EliRibble** [comment]
Should be good to close when merged, yeah

**Comment by rallytime** [comment]
Thanks!
